### PR TITLE
Integrate civic 0.4.13

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will add:

- Civic 0.4.13 integration
- Basic styling of the civic button
- renamed REACT_APP_CLUSTER to NEXT_PUBLIC_REACT_APP_CLUSTER as otherwise nextjs cant read it.

@BouweCeunen I'm assuming that the env var rename has some consequences for our infra?

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -136,7 +136,7 @@ jobs:
             ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-
       - name: Build app
         run: |
-          NEXT_PUBLIC_BASE_PATH= yarn build
+          NEXT_PUBLIC_BASE_PATH= NEXT_PUBLIC_REACT_APP_CLUSTER=devnet yarn build
```
Comment (RuvendeGroote) [inline .github/workflows/cicd.yaml]: remove NEXT_PUBLIC_BASE_PATH

File: .github/workflows/cicd.yaml [hunk 2]
```diff
@@ -199,7 +199,7 @@ jobs:
             ${{ runner.os }}-nextjs-${{ hashFiles('**/package-lock.json') }}-
       - name: Build app
         run: |
-          NEXT_PUBLIC_BASE_PATH= yarn build
+          NEXT_PUBLIC_BASE_PATH= NEXT_PUBLIC_REACT_APP_CLUSTER=mainnet yarn build
```
Comment (RuvendeGroote) [inline .github/workflows/cicd.yaml]: same

**Comment by RuvendeGroote** [comment]
investigate the many rerenders and  duplicate rpc calls

**Comment by RuvendeGroote** [comment]
add info about targetting the right cluster to the readme.
add info about that we use the civic pass (only on devnet) to readme

File: src/components/Identity.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { PublicKey } from "@solana/web3.js";
+import { useConnection, useWallet } from "@solana/wallet-adapter-react";
+import { GatewayProvider } from "@civic/solana-gateway-react";
+import { CivicHeaderSection } from "@components/CivicHeaderSection";
+import { config } from "../config";
+import { useCredixClient } from "@credix/credix-client";
+import { defaultMarketplace } from "../consts";
+
+export const Identity = () => {
```
Comment (RuvenSalamon) [inline src/components/Identity.tsx:10]: Rename to civicprovider (same principle as you did with the clientprovider)

Comment (RuvenSalamon) [inline src/components/Identity.tsx:10]: rename to CivicButton

File: src/pages/index.tsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ const Overview: NextPage = () => {
 
 	useEffect(() => {
 		getMarket();
-	}, [getMarket]);
+	}, []);
```
Comment (RuvendeGroote) [inline src/pages/index.tsx]: add a comment explaining why we ignore the eslint warning here and why getmarket should not be added as a dependency again

File: src/config.ts [hunk 5]
```diff
@@ -12,23 +13,26 @@ const localnetConfig: ClusterConfig = {
 	...baseConfig,
 	name: SolanaCluster.LOCALNET,
 	RPCEndpoint: RPCEndpoint.LOCALNET,
+	stage: CivicStage.LOCALNET,
```
Comment (RuvendeGroote) [inline src/config.ts]: this doesn't need to be part of the config. it's very specific for a particular component and can easily be derived from the cluster in the config already.

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by BouweCeunen — DISMISSED** [review]

File: src/components/IdentityButton.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import { SolanaCluster } from "@credix_types/solana.types";
+
+export const IdentityButton = () => {
+	const wallet = useWallet();
+	const client = useCredixClient();
+	const { connection } = useConnection();
+	const [gatekeeperNetwork, setGatekeeperNetwork] = useState<PublicKey>();
+
+	const getGatekeeperNetwork = useCallback(async () => {
+		const market = await client.fetchMarket(defaultMarketplace);
```
Comment (RuvendeGroote) [inline src/components/IdentityButton.tsx]: can we use the market from the state here?

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Review by BouweCeunen — APPROVED** [review]

**Pull request merged**